core/services/ocr2/plugins/ocr2keeper/evm/registry.go

```diff
@@ -553,30 +558,25 @@ func (r *EvmRegistry) checkUpkeeps(ctx context.Context, keys []types.UpkeepKey)
 				hexutil.EncodeBig(opts.BlockNumber),
 			},
 			Result: &result,
-		}
+		})
 
-		checkResults[i] = &result
+		checkResults = append(checkResults, &result)
 	}
 
 	if err := r.client.BatchCallContext(ctx, checkReqs); err != nil {
 		return nil, err
 	}
 
-	var (
-		multiErr error
-		results  = make([]types.UpkeepResult, len(keys))
-	)
-
 	for i, req := range checkReqs {
 		if req.Error != nil {
 			r.lggr.Debugf("error encountered for key %s with message '%s' in check", keys[i], req.Error)
 			multierr.AppendInto(&multiErr, req.Error)
 		} else {
-			var err error
-			results[i], err = r.packer.UnpackCheckResult(keys[i], *checkResults[i])
+			result, err := r.packer.UnpackCheckResult(keys[i], *checkResults[i])
 			if err != nil {
 				return nil, err
 			}
```

Comment:
similar treatment of multiErr?

Reply:
It looks like a rare case, but if an unpack error occurs it does look like the behavior is just the same as before where we return a nil result and error. This should probably not be an if else.

-> if req.Error: add to multierr & continue
-> if unpack err: add to multierr & continue
-> do stuff with result ...